test262-to-wpt

```diff
@@ -0,0 +1,404 @@
+describes several properties of the test, and a body, which defines the proper 
+test itself.
+
+The test header is defined in YAML and contains properties such as esid (test ID),
+description, features, flags, etc. Some of these properties are relevant to know
+what harnessing libraries to import from the test262 suite (features), whether
+the test should only be tested in strict mode or non-strict mode (flags), etc
+
+Some of the beforementioned properties would need to get serialized to the 
+resulting web-platform-test as a JSON object.
+
+The parser also fetches the body of the test.
+'''
+# Parses a Test262 test. 
+class Test262Parser(object):
+
+    def __init__(self, text):
+        match = re.search('---\*/', text)
+        if match:
+            self.header = text[:match.end(0)]
+            self.body = text[match.end(0)+1:]
+        else:
+            self.header = ""
+            self.body = text
+        self.attrs = {}
+        self.parse_header()
+
+    def parse_header(self):
+        if len(self.header) == 0:
+            return
+        value = []
+        last_attr = None
+        for line in self.header.split('\n'):
+            # End of the attribute section?
+            if re.search(r'---\*/', line):
+                self.store_attribute(last_attr, "\n".join(value))
+                break
+
+            # Attribute? (attr: <text>)
+            match = re.search(r'^\s*(\w+):', line)
+            if match:
+                attr = match.group(1)
+                # Name of attribute changed?
+                if attr != last_attr:
+                    # Should store attribute?
+                    if last_attr is not None:
+                        self.store_attribute(last_attr, "".join(value))
+                    match = re.search(': (.*)\n?', line)
+                    if match:
+                        value = [match.group(1)]
+                    else:
+                        value = []
+                    last_attr = attr
+            else:
+                value.append(line)
+        self.store_attribute(last_attr, "".join(value))
+
+    def store_attribute(self, name, value):
+        if name in ['includes', 'flags', 'features']:
+            self.attrs[name] = self.parse_list(value)
+        else:
+            self.attrs[name] = value
+
+    '''
+    There are two ways of describing a list of properties:
+       attr: [prop1, prop2, ..., propn]
+       attr:
+         - prop1
+         - prop2
+         - ...
+         - propn
+    '''
+    def parse_list(self, text):
+        match = re.search("\[", text)
+        if match:
+            return re.findall("[^\[, \]\n]+", text)
+        else:
+            return re.findall("[^- \n]+", text)
+
+###
+
+# TODO: Put everything into one single template?
+HEADER = trim('''
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="utf-8">
+    <title>###TITLE###</title>
+    <meta name="help" href="https://storage.spec.whatwg.org/#dom-storagemanager-persisted">
```

Comment:
I don't think you meant to include this?

Reply:
Right, likely a copy-paste error.